src/Libraries/Microsoft.Extensions.AI.Evaluation.NLP/Common/SmoothingFunction.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Extensions.AI.Evaluation.NLP.Common;
+
+/// <summary>
+/// Implementations of smoothing functions for BLEU scores taken from
+/// `A Systematic Comparison of Smoothing Techniques for Sentence-Level BLEU` 
+/// by Chen and Cherry. http://acl2014.org/acl2014/W14-33/pdf/W14-3346.pdf.
+/// </summary>
+internal static class SmoothingFunction
+{
+    /// <summary>
+    /// This is the baseline method, which does not apply any smoothing.
+    /// </summary>
+    /// <param name="precisions">N precision values to be smoothed.</param>
+    /// <param name="hypLen">Number of tokens in the hypothesis.</param>
+    /// <returns>Smoothed precision values.</returns>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE0060:Remove unused parameter", Justification = "Matches expected signature of SmoothingFunction")]
```

Comment:
This seems like an analyzer bug, in that it should be taking into account usage sites. Might be worth opening an issue against dotnet/roslyn if there isn't one already. It's possible it'll be won't-fixed because such analysis turns a local problem into a project-wide one.